File: skipad-collector/src/com/skipad/collector/entities/Account.java
package com.skipad.collector.entities;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import com.skipad.collector.common.IAccount;
import com.skipad.collector.common.IProperty;

public class Account implements IAccount {
	
	private long id;
	private String name;
	private String externalId;
	private int ownerId;
	private Date createDate;
	
	private List<IProperty> properties = new ArrayList<>();
	
	public Account(ResultSet reader) throws SQLException {
		id = reader.getLong(ColumnNames.AccountId);
		name = reader.getString(ColumnNames.AccountName);
		externalId = reader.getString(ColumnNames.AccountExternalId);
		ownerId = reader.getInt(ColumnNames.AccountOwnerId);
		createDate = reader.getDate(ColumnNames.AccountCreateDate);
	}

	public void addProperty(IProperty property) {
		properties.add(property);
	}

	@Override
	public long getId() {
		return id;
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public String getExternalId() {
		return externalId;
	}

	@Override
	public int getOwnerId() {
		return ownerId;
	}

	@Override
	public Date getCreateDate() {
		return createDate;
	}

	@Override
	public List<IProperty> getProperties() {
		return properties;
	}

}

File: Skipad/Entities/Account.cs
using System;
using System.Data;
using Inqwise.Skipad.Common;

namespace Inqwise.Skipad.Entities
{
    public class Account : IAccount
    {
        public static class ColumnNames
        {
            public const string Name = "AccountName";
            public const string AccountId = "AccountId";
            public const string ExternalId = "AccountExternalId";
            public const string CreateDate = "CreateDate";
            public const string IsDefault = "IsDefault";
            public const string PackageId = "AccountPropertyPackageId";
        }

        public Account(IDataReader reader)
        {
            Id = Convert.ToInt32(reader[ColumnNames.AccountId]);
            ExternalId = reader[ColumnNames.ExternalId].ToString();
            Name = reader[ColumnNames.Name].ToString();
            CreateDate = Convert.ToDateTime(reader[ColumnNames.CreateDate]);
            var objIsDefault = reader[ColumnNames.IsDefault];
            if (!Convert.IsDBNull(objIsDefault))
            {
                IsDefault = Convert.ToBoolean(objIsDefault); 
            }
            var objPackageId = reader[ColumnNames.PackageId];
            if (!Convert.IsDBNull(objPackageId))
            {
                PackageId = Convert.ToInt32(objPackageId);
            }
        }

        public long Id { get; private set; }
        public string ExternalId { get; private set; }
        public string Name { get; private set; }
        public DateTime CreateDate { get; private set; }
        public bool? IsDefault { get; private set; }
        public int? PackageId { get; private set; }
    }
}